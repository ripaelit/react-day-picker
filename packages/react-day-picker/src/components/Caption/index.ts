export * from './Caption';
export * from './getCaptionProps';
export * from './types';
