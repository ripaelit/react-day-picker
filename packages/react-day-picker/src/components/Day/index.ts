export * from './Day';
export * from './utils/getModifiersStatus';
