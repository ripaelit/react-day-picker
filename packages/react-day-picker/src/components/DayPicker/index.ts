export * from './DayPicker';
// export * from './DayPickerContext';
// export * from './NavigationContext';
