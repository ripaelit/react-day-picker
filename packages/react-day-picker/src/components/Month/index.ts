export * from './Month';
export * from './types';
