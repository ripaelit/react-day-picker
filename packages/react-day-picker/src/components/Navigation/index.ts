export * from './getNavigationProps';
export * from './Navigation';
export * from './types';
export * from './utils';
