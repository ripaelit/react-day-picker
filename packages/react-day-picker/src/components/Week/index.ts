export { Week as WeekRow } from './Week';
export * from './types';
