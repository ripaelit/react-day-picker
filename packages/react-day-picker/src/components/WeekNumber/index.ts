export * from './WeekNumber';
export * from './types';
