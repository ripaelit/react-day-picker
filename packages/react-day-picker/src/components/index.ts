export * from './Caption';
export * from './Day';
export * from './DayPicker';
export * from './Dropdown';
export * from './Head';
export * from './IconDropdown';
export * from './IconNext';
export * from './IconPrev';
export * from './MonthsDropdown';
export * from './Root';
export * from './Row';
export * from './Table';
export * from './WeekNumber';
export * from './YearsDropdown';
