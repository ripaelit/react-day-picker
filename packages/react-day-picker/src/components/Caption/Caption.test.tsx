import React from 'react';

import { RenderResult } from '@testing-library/react';
import tk from 'timekeeper';

import { Caption, CaptionProps } from 'components';
import { customRender } from 'test';

const FrozenDate = new Date(2020, 5);

beforeEach(() => tk.freeze(FrozenDate));
afterEach(() => tk.reset());

const setup = (props?: CaptionProps): RenderResult => {
  return customRender(
    <Caption displayMonth={new Date()} displayIndex={1} {...props} />
  );
};

test('should render correctly', () => {
  const { container } = setup();
  expect(container.firstChild).toMatchSnapshot();
});

describe('when navigation is disabled', () => {
  test.todo('should display the caption label');
  test.todo('should not render the drop-downs');
  test.todo('should not render the navigation');
});

describe('when navigation is enabled', () => {
  describe('when the caption layout is "drop-down"', () => {
    test.todo('should apply the `caption_drop-downs` class name');
    test.todo('should apply the `caption_drop-downs` style');
    test.todo('should render the months drop-down');
    test.todo('should render the years drop-down');
    describe('when a month is selected from the month drop-down', () => {
      test.todo('should go to the selected month');
      test.todo('should cal the `onMonthChange` callback');
    });
    describe('when a year is selected from the year drop-down', () => {
      test.todo('should go to the selected month');
      test.todo('should cal the `onMonthChange` callback');
    });
  });
  describe('when the caption layout is "buttons"', () => {
    test.todo('should display the caption label');
    test.todo('should display the navigation');
    describe('when rendering multiple months', () => {
      describe('if is not the last month', () => {
        test.todo('should hide the next button');
      });
      describe('if is not the first month', () => {
        test.todo('should hide the previous button');
      });
    });
    describe('when clicking the previous button', () => {
      test.todo('should go to the previous month');
      test.todo('should call the `onMonthChange` callback');
      describe('if no previous month to navigate', () => {
        test.todo('should not change the month');
        test.todo('should not call the `onMonthChange` callback');
      });
      describe('if no next month to navigate', () => {
        test.todo('should not change the month');
        test.todo('should not call the `onMonthChange` callback');
      });
    });
    describe('when clicking the next button', () => {
      test.todo('should go to the next month');
      test.todo('should call the `onMonthChange` callback');
      describe('if no next month to navigate', () => {
        test.todo('should not change the month');
        test.todo('should not call the `onMonthChange` callback');
      });
      describe('if no next month to navigate', () => {
        test.todo('should not change the month');
        test.todo('should not call the `onMonthChange` callback');
      });
    });
  });
});

tk.reset();
