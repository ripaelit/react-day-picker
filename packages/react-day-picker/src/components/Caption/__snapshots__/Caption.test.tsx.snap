// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  class="rdp-caption"
  role="caption"
>
  <div
    aria-live="polite"
    class="rdp-caption_label"
  >
    June 2020
  </div>
  <div
    class="rdp-nav"
  >
    <button
      aria-label="Go to previous month: May 2020"
      class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_previous"
      type="button"
    >
      <svg
        class="rdp-nav_icon"
        height="16px"
        viewBox="0 0 120 120"
        width="16px"
      >
        <path
          d="M69.490332,3.34314575 C72.6145263,0.218951416 77.6798462,0.218951416 80.8040405,3.34314575 C83.8617626,6.40086786 83.9268205,11.3179931 80.9992143,14.4548388 L80.8040405,14.6568542 L35.461,60 L80.8040405,105.343146 C83.8617626,108.400868 83.9268205,113.317993 80.9992143,116.454839 L80.8040405,116.656854 C77.7463184,119.714576 72.8291931,119.779634 69.6923475,116.852028 L69.490332,116.656854 L18.490332,65.6568542 C15.4326099,62.5991321 15.367552,57.6820069 18.2951583,54.5451612 L18.490332,54.3431458 L69.490332,3.34314575 Z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </svg>
    </button>
    <button
      aria-label="Go to next month: July 2020"
      class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_previous"
      type="button"
    >
      <svg
        class="rdp-nav_icon"
        height="16px"
        viewBox="0 0 120 120"
        width="16px"
      >
        <path
          d="M49.8040405,3.34314575 C46.6798462,0.218951416 41.6145263,0.218951416 38.490332,3.34314575 C35.4326099,6.40086786 35.367552,11.3179931 38.2951583,14.4548388 L38.490332,14.6568542 L83.8333725,60 L38.490332,105.343146 C35.4326099,108.400868 35.367552,113.317993 38.2951583,116.454839 L38.490332,116.656854 C41.5480541,119.714576 46.4651794,119.779634 49.602025,116.852028 L49.8040405,116.656854 L100.804041,65.6568542 C103.861763,62.5991321 103.926821,57.6820069 100.999214,54.5451612 L100.804041,54.3431458 L49.8040405,3.34314575 Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;
